cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureUtils.java

```diff
@@ -300,41 +302,78 @@ private void deallocateInstance(AuthenticatedContext ac, List<CloudVmInstanceSta
     @Retryable(backoff = @Backoff(delay = 1000, multiplier = 2, maxDelay = 10000), maxAttempts = 5)
     public List<CloudVmInstanceStatus> deleteInstances(AuthenticatedContext ac, List<CloudInstance> vms) {
         LOGGER.info("Delete VMs: {}", vms.stream().map(CloudInstance::getInstanceId).collect(Collectors.toList()));
-        List<CloudVmInstanceStatus> statuses = new ArrayList<>();
-        List<Completable> deleteCompletables = new ArrayList<>();
-        for (CloudInstance vm : vms) {
-            String resourceGroupName = azureResourceGroupMetadataProvider.getResourceGroupName(ac.getCloudContext(), vm);
-            AzureClient azureClient = ac.getParameter(AzureClient.class);
-            deleteCompletables.add(azureClient.deleteVirtualMachine(resourceGroupName, vm.getInstanceId())
-                    .doOnError(throwable -> {
-                        LOGGER.error("Error happened on azure instance delete: {}", vm, throwable);
-                        statuses.add(new CloudVmInstanceStatus(vm, InstanceStatus.FAILED, throwable.getMessage()));
-                    })
-                    .doOnCompleted(() -> statuses.add(new CloudVmInstanceStatus(vm, InstanceStatus.TERMINATED)))
-                    .subscribeOn(Schedulers.io()));
+        List<CloudVmInstanceStatus> actualVmStatuses = new ArrayList<>();
+        List<CloudVmInstanceStatus> statusesAfterDelete = new ArrayList<>();
```

Comment:
For me deletedVmStatuses means all of the vm is already deleted, therefore their status can only be TERMINATED. For example if something goes wrong and one of the vm is not deleted it would be confusing. I think statusesAfterDelete expresses better that "what is their status after we tried to delete them"